File: thorough training/arrays/alien_order.php
<?php
class Solution {

    /**
     * @param String[] $words
     * @param String $order
     * @return Boolean
     */
    function isAlienSorted($words, $order) {
        $rank = [];

        for($i = 0; $i < strlen($order); $i++) {
            $rank[$order[$i]] = $i;
        }

        for($w = 0; $w < count($words) - 1; $w++) {
            $current_word = $words[$w];
            $next_word = $words[$w + 1];

            $currlength = strlen($current_word);
            $nextlength = strlen($next_word);
            $minlength = min($currlength, $nextlength);

            for($minword = 0; $minword < $minlength; $minword++) {
                $frstletter = $current_word[$minword];
                $nextletter = $next_word[$minword];

                if($rank[$frstletter] < $rank[$nextletter]) {
                    break;
                } elseif($rank[$frstletter] > $rank[$nextletter]) {
                    return false;
                }
            }

            if(($minword == $minlength) && ($currlength > $nextlength)) {
                return false;
            }
        }

        return true;
    }
}

File: thorough training/arrays/alien_order.py
def isAlienSorted(words, order):
    rank = {char: i for i, char in enumerate(order)}

    for i in range(len(words) - 1):
        word1 = words[i]
        word2 = words[i + 1]
        min_len = min(len(word1), len(word2))

        j = 0
        while j < min_len:
            c1 = word1[j]
            c2 = word2[j]
            if rank[c1] < rank[c2]:
                break   
            elif rank[c1] > rank[c2]:
                return False
            j += 1

        if j == min_len and len(word1) > len(word2):
            return False

    return True
